### PR TITLE
Rework --help for `deploy` command 

### PR DESCRIPTION
**Pull request by pmatulis**
## Description of change

Improving the Examples section for the `deploy` command was my original goal here but I ended up going over all the text as well. I cut out one `--device` example as it was for providing general syntax (the documentation should handle that).

There is a lack of consistency in how the Examples section is formatted in many Juju commands. I would like to offer up the Examples section in this PR as a standard. A few commands already use it. Sample:

```
Examples:

Deploy to a new machine:

    juju deploy apache2

Deploy to machine 23:

    juju deploy mysql --to 23

Deploy to a new LXD container on a new machine:

    juju deploy mysql --to lxd
```  

## QA steps

View the help (`juju help deploy`) and see the revised output.

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: cmd/juju/application/deploy.go [hunk 1]
```diff
@@ -387,80 +387,76 @@ type DeployCommand struct {
...
-unexpected behavior. 
+  juju deploy /path/to/bundle.yaml
 
-Local bundles are specified with a direct path to a bundle.yaml file.
-For example:
+When charms that include LXD profiles (to override configurations or devices
+when creating containers) are deployed the profiles are validated using a
+hardcoded allow/deny list. This check can be bypassed with the use of the
+'--force' option. However, doing so is not recommended as it can lead to
+unexpected behaviour.
```
Comment (SimonRichardson) [inline cmd/juju/application/deploy.go]: :+1:

**Comment by pmatulis** [comment]
[Improve help for `deploy` command](https://trello.com/c/wZVBxlSk/721-improve-help-for-deploy-command)

File: cmd/juju/application/deploy.go [hunk 2]
```diff
@@ -387,80 +387,76 @@ type DeployCommand struct {
 }
 
 const deployDoc = `
-<charm or bundle> can be a charm/bundle URL, or an unambiguously condensed
-form of it; assuming a current series of "trusty", the following forms will be
-accepted:
+The charm (or bundle) can be referred to by a charm/bundle URL, or an
+unambiguously condensed form of it. Assuming a current series of "xenial", the
```
[the remark this answers is not in the document]
Comment (pmatulis) [reply cmd/juju/application/deploy.go]: I don't remember. :)
Lemme chew on it.

File: cmd/juju/application/deploy.go [hunk 3]
```diff
@@ -387,80 +387,76 @@ type DeployCommand struct {
...
 
 If '--series' is not specified, the charm's default series is used. The default
-series for a charm is the first one specified in the charm metadata. If the
-specified series is not supported by the charm, this results in an error,
-unless '--force' is used.
+series is the top-most one specified in the charm metadata. If the specified
+series is not supported by the charm an error is emitted. The '--force' option
+can be used to override this check:
 
   juju deploy /path/to/charm --series wily --force
```
[the remark this answers is not in the document]
Comment (pmatulis) [reply cmd/juju/application/deploy.go]: Ohh, I thought I up'd code names to Xenial. Thanks.

File: cmd/juju/application/deploy.go [hunk 4]
```diff
@@ -469,105 +465,126 @@ use
...
   juju deploy foo --resource bar=/some/file.tgz --resource baz=./docs/cfg.xml
 
-Where 'bar' and 'baz' are resources named in the metadata for the 'foo' charm.
+Where 'bar' and 'baz' are named in the metadata file for charm 'foo'.
 
-When using a placement directive to deploy to an existing machine or container
-('--to' option), the ` + "`juju status`" + ` command should be used for guidance. A few
-placement directives are provider-dependent (e.g.: 'zone').
+Use the '--to' option to deploy to an existing machine or container by
+specifying a "placement directive". The ` + "`status`" + ` command should be used for
```
[the remark this answers is not in the document]
Comment (pmatulis) [reply cmd/juju/application/deploy.go]: Having to read "the `juju <subcommand>` command" everywhere is taxing after just a short while given how many references to commands are made throughout the docs. I do often employ "do this with juju <subcommand>" though.

My docs policy is to only refer to just the part after "juju" when I refer to a "command". This is often true in the command help as well. I phase out the other way when I come across them.

I give the average reader credit that they understand that, given the context, it refers to a Juju command.

File: cmd/juju/application/deploy.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pmatulis) [reply cmd/juju/application/deploy.go]: Figured this out on #juju.

**Comment by pmatulis** [comment]
I got some conflicts (due to using the online editor I think) and I opened #9358 instead. Closing this one.

**Pull request closed without merging**